pgmpy/base/DAG.py

```diff
@@ -580,12 +564,17 @@ def is_dconnected(self, start, end, observed=None):
         >>> student.is_dconnected('grades', 'sat')
         True
         """
-        if end in self.active_trail_nodes(start, observed)[start]:
+        if (
+            end
+            in self.active_trail_nodes(
+                variables=start, observed=observed, include_latents=include_latents
+            )[start]
+        ):
             return True
         else:
             return False
 
-    def minimal_dseparator(self, start, end):
+    def minimal_dseparator(self, start, end, include_latents=False):
```

Comment:
I don't think we want to add the `include_latents` argument here either. I think we always want the separating set to be observed (to be usable in practice). I don't think this would have any practical use case and would be confusing for users.

Reply:
@ankurankan I have one question. This may be because I am a beginner in Bayesian inference. But if we always want the separating set to be observed, why do we expect the get_independencies to return an Independencies object representing X and Z are independent given Y in test_BayesianModel.py line 230. ? 

```python
fork = BayesianModel([("Y", "X"), ("Y", "Z")], latents=["Y"])
self.assertEqual(
            fork.get_independencies(include_latents=True),
            Independencies(("X", "Z", "Y")),
        )
```

Further, this exact test case fail when I remove include_latents from minimal_dseparator as Y is removed from the separator for being latent and resulting the separator set  to be empty which then cause the minimal_separator to return None and hence the test case fail.

```python
# If any of the parents were latents, take the latent's parent
while len(separator.intersection(self.latents)) != 0:
           separator_copy = separator.copy()
            for u in separator:
                if u in self.latents:
                    separator_copy.remove(u)
                    separator_copy.update(set(self.predecessors(u)))
            separator = separator_copy

        # Remove the start and end nodes in case it reaches there while removing latents.
        separator.difference_update({start, end})

        # If the initial set is not able to d-separate, no d-separator is possible.
        if an_graph.is_dconnected(
            start, end, observed=separator
        ):
            return None
```

Reply:
@ankurankan Regarding my above question, as the latent are removed as candidate for minimal_dseparator, and the test case expect to return X and Z are independent given Y (latent), how do we handle this specific condition ? Should we just return Independencies instance with the assertion that x and y are independent given all latent variable or only latent variable which are parent of x or y or latent variable with path from x or y ? Can you please help in this solution.

Reply:
@Nuna7 Sorry for the late reply. On second thought, we can keep the `include_latents` argument for both methods. I still can't think of any use case for it though, but I think there's no harm in having that option with the default being set to False.

Reply:
@ankurankan I add the include_latents and its documentation back. Anymore changes ?